lib/utils/fdt/fdt_helper.c

```diff
@@ -401,7 +401,7 @@ static int fdt_parse_isa_one_hart(const char *isa, unsigned long *extensions)
 				continue;			\
 			}
 
-		for (j = 0; j < SBI_HART_EXT_MAX; j++) {
+		for (j = 0; j < sbi_hart_ext_size; j++) {
```

Comment:
Would this then clash with other iterations relying on `sbi_hart_ext` being that previous size?

E.g. just further down:
```C
	for (i = 0; i < BITS_TO_LONGS(SBI_HART_EXT_MAX); i++)
		extensions[i] |= hart_exts[i];
	return 0;
```

Can the creation of `sbi_hart_ext` be changed to have something like null entries? I don't really know C, would do this much different in Rust. :crab: Maybe just changing all the code would be better.

Reply:
yeah, in a modern language this would be quite different. I believe that the code you point to is unaffected, all this is doing is fixing an iterator.

Reply:
it lost my response. I believe that's ok, the issue here is the sbi_hart_ext array, not hart_exts or extensions.

Reply:
i.e. extensions is this
unsigned long extensions[BITS_TO_LONGS(SBI_HART_EXT_MAX)];
hart_exts is this

        hart_exts = sbi_scratch_offset_ptr(scratch, fdt_isa_bitmap_offset);

I think both are safe.